File: app/src/main/java/org/stepic/droid/storage/structure/DbStructureLesson.java
package org.stepic.droid.storage.structure;

@Deprecated
public final class DbStructureLesson {
    public static final String LESSONS = "lessons";

    public static final class Column {
        @Deprecated
        public static final String ID = "_id";
        public static final String LESSON_ID = "lesson_id";
        public static final String STEPS = "steps_arr";
        public static final String IS_FEATURED = "is_featured";
        public static final String IS_PRIME = "is_prime";
        public static final String PROGRESS = "progress";
        public static final String OWNER = "owner";
        public static final String SUBSCRIPTIONS = "subscriptions";
        public static final String VIEWED_BY = "viewed_by";
        public static final String PASSED_BY = "passed_by";
        public static final String DEPENDENCIES = "dependencies";
        public static final String IS_PUBLIC = "is_public";
        public static final String TITLE = "title";
        public static final String SLUG = "slug";
        public static final String CREATE_DATE = "create_date";
        public static final String UPDATE_DATE = "update_date";
        public static final String LEARNERS_GROUP = "learners_group";
        public static final String TEACHER_GROUP = "teacher_group";
        public static final String IS_CACHED = "is_cached";
        public static final String IS_LOADING = "is_loading";
        public static final String COVER_URL = "cover_url";
        public static final String VOTE_DELTA = "vote_delta";

    }

}

File: app/src/main/java/org/stepik/android/cache/lesson/structure/DbStructureLesson.kt
package org.stepik.android.cache.lesson.structure

import androidx.sqlite.db.SupportSQLiteDatabase

object DbStructureLesson {
    const val TABLE_NAME = "lesson"

    object Columns {
        const val ID = "id"
        const val TITLE = "title"
        const val SLUG = "slug"
        const val COVER_URL = "cover_url"
        const val COURSES = "courses"
        const val STEPS = "steps"
        const val ACTIONS = "actions"
        const val IS_FEATURED = "is_featured"
        const val PROGRESS = "progress"
        const val OWNER = "owner"
        const val SUBSCRIPTIONS = "subscriptions"
        const val VIEWED_BY = "viewed_by"
        const val PASSED_BY = "passed_by"
        const val VOTE_DELTA = "vote_delta"
        const val LANGUAGE = "language"
        const val IS_PUBLIC = "is_public"
        const val CREATE_DATE = "create_date"
        const val UPDATE_DATE = "update_date"
        const val LEARNERS_GROUP = "learners_group"
        const val TEACHERS_GROUP = "teachers_group"
        const val TIME_TO_COMPLETE = "time_to_complete"
    }

    fun createTable(db: SupportSQLiteDatabase) {
        db.execSQL("""
            CREATE TABLE IF NOT EXISTS $TABLE_NAME (
                ${DbStructureLesson.Columns.ID} LONG PRIMARY KEY,
                ${DbStructureLesson.Columns.TITLE} TEXT,
                ${DbStructureLesson.Columns.SLUG} TEXT,
                ${DbStructureLesson.Columns.COVER_URL} TEXT,
                ${DbStructureLesson.Columns.STEPS} TEXT,
                ${DbStructureLesson.Columns.IS_FEATURED} INTEGER,
                ${DbStructureLesson.Columns.PROGRESS} TEXT,
                ${DbStructureLesson.Columns.OWNER} LONG,
                ${DbStructureLesson.Columns.SUBSCRIPTIONS} TEXT,
                ${DbStructureLesson.Columns.VIEWED_BY} LONG,
                ${DbStructureLesson.Columns.PASSED_BY} LONG,
                ${DbStructureLesson.Columns.VOTE_DELTA} LONG,
                ${DbStructureLesson.Columns.LANGUAGE} TEXT,
                ${DbStructureLesson.Columns.IS_PUBLIC} INTEGER,
                ${DbStructureLesson.Columns.CREATE_DATE} LONG,
                ${DbStructureLesson.Columns.UPDATE_DATE} LONG,
                ${DbStructureLesson.Columns.LEARNERS_GROUP} TEXT,
                ${DbStructureLesson.Columns.TEACHERS_GROUP} TEXT,
                ${DbStructureLesson.Columns.TIME_TO_COMPLETE} LONG
            )
        """.trimIndent())
    }
}